idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionPropsFileType.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import com.intellij.openapi.fileTypes.FileTypes;
+import com.intellij.openapi.fileTypes.LanguageFileType;
+import javax.swing.Icon;
+
+public final class VersionPropsFileType extends LanguageFileType {
+
+    public static final VersionPropsFileType INSTANCE = new VersionPropsFileType();
+
+    private VersionPropsFileType() {
+        super(VersionPropsLanguage.INSTANCE);
+    }
+
+    @Override
+    public String getName() {
+        return "VersionProps File";
+    }
+
+    @Override
+    public String getDescription() {
+        return "VersionProps language file";
+    }
+
+    @Override
+    public String getDefaultExtension() {
+        return "props";
```

Comment:
I wonder if results in our plugin being used on all `props` files? Or do we constrain it to working on just `versions.props` somehow.

Reply:
Yes it did, the fix was in `plugin.xml` where `fileType` needed passing `filesNames` rather than an extension

Reply:
Ive also added a test to ensure this is not regressed